### PR TITLE
Fix --local-rank argument compatibility with torchrun

### LINKED ISSUE #153: local-rank argument issue
python -m torch.distributed.launch --nproc_per_node=6 train.py --checkpoint ./pretrained_checkpoint/sam_vit_l_0b3195.pth --model-type vit_l --output work_dirs/hq_sam_l
/home/azuryl/anaconda3/envs/sam_hq2/lib/python3.10/site-packages/torch/distributed/launch.py:208: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use-env is set by default in torchrun.
If your script expects --local-rank argument to be set, please
change it to read from os.environ['LOCAL_RANK'] instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

  main()
W0306 18:12:40.932000 150874 site-packages/torch/distributed/run.py:792]
W0306 18:12:40.932000 150874 site-packages/torch/distributed/run.py:792] *****************************************
W0306 18:12:40.932000 150874 site-packages/torch/distributed/run.py:792] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
W0306 18:12:40.932000 150874 site-packages/torch/distributed/run.py:792] *****************************************
usage: HQ-SAM --output OUTPUT [--model-type MODEL_TYPE] --checkpoint CHECKPOINT [--device DEVICE] [--seed SEED] [--learning_rate LEARNING_RATE] [--start_epoch START_EPOCH]
              [--lr_drop_epoch LR_DROP_EPOCH] [--max_epoch_num MAX_EPOCH_NUM] [--input_size INPUT_SIZE] [--batch_size_train BATCH_SIZE_TRAIN] [--batch_size_valid BATCH_SIZE_VALID]
              [--model_save_fre MODEL_SAVE_FRE] [--world_size WORLD_SIZE] [--dist_url DIST_URL] [--rank RANK] [--local_rank LOCAL_RANK] [--find_unused_params] [--eval] [--visualize]
              [--restore-model RESTORE_MODEL]
HQ-SAM: error: unrecognized arguments: --local-rank=5

should modified to  torchrun --nproc_per_node=8 train.py --checkpoint ./pretrained_checkpoint/sam_vit_l_0b3195.pth --model-type vit_l --output work_dirs/hq_sam_l

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Summary
- Accept both `--local-rank` (hyphen) and `--local_rank` (underscore) for the distributed training argument
- Fixes compatibility with PyTorch's `torchrun` which passes `--local-rank` instead of `--local_rank`

## Problem
When using `torchrun` for distributed training:
```bash
torchrun --nproc_per_node=6 train.py ...
```

The script fails with:
```
HQ-SAM: error: unrecognized arguments: --local-rank=5
```

This happens because `torchrun` passes `--local-rank` (with hyphen), but the script only accepted `--local_rank` (with underscore).

## Solution
Changed the argument definition to accept both forms using argparse's multiple option names feature.

## Test plan
- [ ] Run distributed training with `torchrun --nproc_per_node=N train.py`
- [ ] Verify backward compatibility with `torch.distributed.launch` (deprecated)

Fixes #153

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
bump — this fixes the --local-rank argument to work with torchrun's --local_rank convention. otherwise it crashes on launch. lmk if anything needs changing

**Comment by Mr-Neutr0n** [comment]
Friendly ping -- this one-line fix in train/train.py resolves the --local-rank vs --local_rank mismatch that causes torchrun to crash on launch. Would appreciate a review when you get a chance.

**Comment by Mr-Neutr0n** [comment]
Closing this one to tidy up my open pull requests.

It's been open around five months with no review activity, and I'd rather withdraw it than leave it sitting in your queue. Nothing needed from you, and no hard feelings at all.

If the fix is still wanted, this can be reopened, or I'm happy to redo it properly against current main. Apologies for the noise.
